students/Josh_HOff/lesson04/trigrams.py

```diff
@@ -0,0 +1,69 @@
+import pathlib
+import os
+import random
+newdict = {}
+words = ''
+b = ''
+count = 0
+#main function
+if __name__ == "__main__":
+    with open('words.txt', "r") as word_list:
+        words = word_list.read().split(' ')
+        temp_list = words[:]
+        #this 
+        for i in temp_list:
+            words[count] = i.replace('\n', ' ').replace('(', '').replace(')', '')
+            if ' ' in words[count]:
+                check = words[count].split(' ')
+                words[count] = check[0]
+                words.insert((count+1), check[1])
+                count += 1
+            count += 1
+
+        #this loop checks for new lines and removes them, seems cleaner
+        count = 0
+        for i in words:
+            if f'{i[:1]}' == '\n':
+                words[count] = i[1:]
+            count += 1
+        #this loop creates the dictionary needed to complete the assignment
+        for x in range(len(words) - 2):
+            key = f'{words[x]} {words[x+1]}'
+            if key in newdict:
+                newdict[key] += [words[x + 2]]
+            else:
+                newdict[key] = [words[x + 2]]
+        #this chooses a random key to start with
+        a = random.choice(list(newdict.keys()))
+        a += f' {newdict[a][0]}'
+        #loops over dictionary until it cannot find words to continue the paragraph
+        while True:
+            b = a.split(' ')
+            b = f'{b[-2]} {b[-1]}'
+            if b not in newdict:
+                break
+            elif len(newdict[b]) > 1:
+                number = random.choice(range(len(newdict[b])))
+                a += f' {newdict[b][number]}'
+                del newdict[b][number]
+            else:
+                number = 0
+                a += f' {newdict[b][number]}'
+#            if b not in newdict:
+#                break
+#            else:
+#                a += f' {newdict[b][number]}'
```

Comment:
recommendation would be to split up the code into smaller functions; what's nice about that is it can pretty much replace your comments with self-descriptive functions names and makes it easier to read the code.